src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -52,7 +52,21 @@ class ReactStateOLXParser {
     const choice = [];
     let compoundhint = [];
     let widget = {};
-    const { answers } = this.problemState;
+    // eslint-disable-next-line prefer-const
+    let { answers, generalFeedback } = this.problemState;
+    // general feedback replaces selected feedback if all inocrrect selected feedback is the same.
+    if (generalFeedback !== ''
+    && answers.every(
+      answer => (
+        answer.correct
+          ? true
+          : answer?.selectedFeedback === answers.find(a => a.correct === false).selectedFeedback
+      ),
+    )) {
+      answers = answers.map(answer => (!answer?.correct
+        ? { ...answer, selectedFeedback: generalFeedback }
+        : answer));
+    }
```

Comment:
If user writes a generalFeedback, then sets one selectedFeedback for an incorrect answer, the generalFeedback would not be saved at all right? I think I would expect the generalFeedback to appear for incorrect answers that didn't have any selectedFeedback filled out.

A niche case: If user copy/pastes the same selectedFeedback into all their incorrect answers (with the intention of changing later), and adds generalFeedback as well, their selectedFeedback would be overridden here right?

I think it would be good if anything entered is saved, so when a course editor saves and reopens the block, it looks exactly the same as they left it.

Please let me know if I'm not understanding this correctly though since there's a good amount of logic happening in these lines.

Reply:
> If user writes a generalFeedback, then sets one selectedFeedback for an incorrect answer, the generalFeedback would not be saved at all right?

Correct.

>I think I would expect the generalFeedback to appear for incorrect answers that didn't have any selectedFeedback filled out.

 general feedback replaces selected feedback if all incorrect selected feedback is the same. In all other cases, it does not.